### ISSUE TITLE
Question about "Active" setting.

### ISSUE THREAD

**Issue by MostHated** [usrbinsam/mini-key-server#3]
Hey there,
    I got this all working but I would like to see if you could clarify something for me. I see that in the modification menu there is an "Active" setting, but it is not shown on the keys information display screen. What I am wondering is, is it possible to use that "Active" setting to manage something similar to a subscription service in that I would not care about how many activations there were or were used, but having the application have a license key and have it check in with the server now and then to make sure the subscription is still activated. If it is not activated then I would have it disable features in my application. 

I was just wondering if you happen to know off hand if the Active setting might be able to be used for that, if not, would it be possible to get a small feature like that added in? Just a periodic check and way to cut a license from being valid anymore. 

Thanks
-MH

**Comment by miniCruzer** [comment]
Yes that is what Active is for. You would create a key with unlimited activations, then set/unset the *Active* status based on the subscription.

I should probably add that to the key detail, however there is an "Active" column on the key list that says Yes or No.

**Comment by MostHated** [comment]
I tried this out, but it only lets you save if the number is either -1 or if the number is > 0, so you can't actually deactivate it because they would always have either unlimited or at least 1. It just keeps giving a message until you change it to one of those values. I didn't look yet, but I would imagine it is not something crazy to have to adjust. 

Thanks though for this, it seems like it is pretty solid. 

One thing I was wondering about though, do you have any recommendations on how to go about encrypting the data when it is being sent back and forth so that someone can't just monitor their own packets and see what the response code is and then just sent it to themselves?

**Comment by miniCruzer** [comment]
https://github.com/miniCruzer/mini-key-server/commit/00800aebf7aad40b0b6a68b9db57f319a1fcb84a should fix the remaining activations however you probably wouldn't want to create a key with 0 activations.

https://github.com/miniCruzer/mini-key-server/commit/a62f8f0a5378150323d1f385551ae8141307aa8c will show the active status for each key and convenience buttons for deactivating them.

The [README](https://github.com/miniCruzer/mini-key-server/blob/master/README.md#implications) talks about security. You should run this software behind a reverse proxy server configured with HTTPS and HPKP to both encrypt and authenticate your activation server, that way it cannot be spoofed.

**Comment by MostHated** [comment]
Awesome, thanks, I will check out there change here shortly. I do apologize, even though I looked through the documentation a number of times, for some reason I guess I never made it quite to the end, so I missed that part on security. 

 I tested out my domain at Qualsys, unfortunately it came back with "This server supports weak Diffie-Hellman (DH) key exchange parameters. Grade capped to B.  "  I am already using Nginx and reverse proxy on a per virtual domain basis via socket and separate PID but I have never had to make any adjustments based on specific security requirements before, so I will have to check and see if that is because I am using LetsEncrypt or a setting in Nginx.
